src/main/java/org/prebid/server/auction/BidResponseCreator.java

```diff
@@ -384,98 +428,60 @@ private static void tryAddWinningBidByBidder(Bid bid, String bidder,
         }
     }
 
-    private static Stream<Bid> getBids(BidderResponse bidderResponse) {
-        return Stream.of(bidderResponse)
-                .map(BidderResponse::getSeatBid)
-                .filter(Objects::nonNull)
-                .map(BidderSeatBid::getBids)
-                .filter(Objects::nonNull)
-                .flatMap(Collection::stream)
-                .map(BidderBid::getBid);
-    }
-
     /**
      * Corresponds cacheId (or null if not present) to each {@link Bid}.
      */
-    private Future<CacheServiceResult> toBidsWithCacheIds(List<BidderResponse> bidderResponses,
-                                                          Set<Bid> bidsToCache,
-                                                          AuctionContext auctionContext,
-                                                          BidRequestCacheInfo cacheInfo,
-                                                          EventsContext eventsContext) {
-
+    private Future<CacheServiceResult> cacheBids(Set<BidInfo> bidsToCache,
+                                                 AuctionContext auctionContext,
+                                                 BidRequestCacheInfo cacheInfo,
+                                                 EventsContext eventsContext) {
         if (!cacheInfo.isDoCaching()) {
             return Future.succeededFuture(CacheServiceResult.of(null, null, toMapBidsWithEmptyCacheIds(bidsToCache)));
         }
 
         // do not submit non deals bids with zero price to prebid cache
-        final List<Bid> bidsValidToBeCached = bidsToCache.stream()
+        final List<BidInfo> bidsValidToBeCached = bidsToCache.stream()
                 .filter(BidResponseCreator::isValidForCaching)
+                // for test consistency
```

Comment:
If this isn't required for production usage, I would recommend removing this step.

Reply:
I couldn't get permanent /cache request order otherwise.

Reply:
Please take a look at https://github.com/prebid/prebid-server-java/blob/3ce7df94f7b6f42a4d4eaa9fb52d10b54f048b93/src/test/java/org/prebid/server/it/IntegrationTest.java#L185. Does this help or at least gives an idea how to make tests reliable?